src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -3743,12 +3744,42 @@ ImmutableArray<PropertySymbol> addProperties(ImmutableArray<ParameterSymbol> rec
                     if (!memberSignatures.TryGetValue(targetProperty, out var existingMember))
```

Comment:
I am wondering if it is possible to simplify this based on the fact that it's not possible to explicitly declare a field and property with the same name in source.

Reply:
It's probably possible, but I wanted to re-use the existing infrastructure (existing dictionary with special comparer, and especially `OverriddenOrHiddenMembersHelpers`)

---
In reply to: [612079277](https://github.com/dotnet/roslyn/pull/52480#discussion_r612079277) [](ancestors = 612079277)